services/ecs.py

```diff
@@ -2,52 +2,226 @@
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+
+        if task_definition.get("networkMode") == "awsvpc":
+            compliant_resources.append(task_definition["taskDefinitionArn"])
+        else:
+            non_compliant_resources.append(task_definition["taskDefinitionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_containers_nonprivileged():
+    compliant_resources = []
+    non_compliant_resources = []
+    task_definitions = client.list_task_definitions(status="ACTIVE")["taskDefinitionArns"]
+    latest_task_definitions = {}
+
+    for task_definition in task_definitions:
+        if latest_task_definitions.get(task_definition.rsplit(":", 1)[0], 0) < int(task_definition.rsplit(":", 1)[1]):
+            latest_task_definitions[task_definition.rsplit(":", 1)[0]] = int(task_definition.rsplit(":", 1)[1])
+
+    for task_definition in latest_task_definitions.keys():
+        task_definition_arn = f"{task_definition}:{latest_task_definitions[task_definition]}"
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+        containers = task_definition["containerDefinitions"]
+
+        for container in containers:
+            if container.get("privileged") == True:
+                non_compliant_resources.append(task_definition["taskDefinitionArn"])
+                break
+        else:
+            compliant_resources.append(task_definition["taskDefinitionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_containers_readonly_access():
+    compliant_resources = []
+    non_compliant_resources = []
+    task_definitions = client.list_task_definitions(status="ACTIVE")["taskDefinitionArns"]
+    latest_task_definitions = {}
+
+    for task_definition in task_definitions:
+        if latest_task_definitions.get(task_definition.rsplit(":", 1)[0], 0) < int(task_definition.rsplit(":", 1)[1]):
+            latest_task_definitions[task_definition.rsplit(":", 1)[0]] = int(task_definition.rsplit(":", 1)[1])
+
+    for task_definition in latest_task_definitions.keys():
+        task_definition_arn = f"{task_definition}:{latest_task_definitions[task_definition]}"
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+        containers = task_definition["containerDefinitions"]
+
+        for container in containers:
+            if container.get("readonlyRootFilesystem") == False:
```

Comment:
`readonlyRootFilesystem` attribute가 존재하지 않는 경우, `None`이 반환되면서 의도한 바와 다르게 처리될 수 있습니다.
```suggestion
            if not container.get("readonlyRootFilesystem"):
```